src/ucs/sys/ucs_netlink.h

```diff
@@ -0,0 +1,88 @@
+/**
+* Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2024. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifndef UCS_NETLINK_H
+#define UCS_NETLINK_H
+
+#include <ucs/type/status.h>
+
+#include <linux/netlink.h>
+#include <netinet/in.h>
+#include <stddef.h>
+
+BEGIN_C_DECLS
+
+
+/**
+ * Callback function for parsing individual netlink messages.
+ *
+ * @param [in] nlh    Pointer to the netlink message header.
+ * @param [in] nl_msg Pointer to the netlink message payload.
+ * @param [in] arg    User-provided argument passed through from the caller.
+ *
+ * @return UCS_INPROGRESS if parsing should continue to the next message,
```

Comment:
Maybe simplify:
`UCS_OK is parsing is complete, UCS_INPROGRESS is there are more messages to be parsed, or error code otherwise\in case of error`